The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/9488044534/job/26146051936) [(plain)](https://github.com/rust-lang/rust/commit/39801b7c31aa054b997f1f4a1e8f868c2a29e4c1/checks/26146051936/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Getting action download info
Download action repository 'msys2/setup-msys2@v2.22.0' (SHA:cc11e9188b693c2b100158c3322424c4cc1dadea)
Download action repository 'actions/checkout@v4' (SHA:a5ac7e51b41094c92402da3b24376905380afc29)
Download action repository 'actions/upload-artifact@v4' (SHA:65462800fd760344b1a7b4382951275a0abb4808)
Complete job name: PR - mingw-check-tidy
git config --global core.autocrlf false
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
---
COPY scripts/sccache.sh /scripts/
RUN sh /scripts/sccache.sh

COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt \
    && pip3 install virtualenv
COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---

#12 [5/8] COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
#12 DONE 0.0s

#13 [6/8] RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt     && pip3 install virtualenv
#13 0.447   Downloading binaryornot-0.4.4-py2.py3-none-any.whl (9.0 kB)
#13 0.461 Collecting boolean-py==4.0
#13 0.466   Downloading boolean.py-4.0-py3-none-any.whl (25 kB)
#13 0.476 Collecting chardet==5.1.0
---
#13 3.743 Building wheels for collected packages: reuse
#13 3.744   Building wheel for reuse (pyproject.toml): started
#13 4.084   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 4.085   Created wheel for reuse: filename=reuse-1.1.0-cp310-cp310-manylinux_2_35_x86_64.whl size=181117 sha256=f5f58750481f69515c2c0d1d503daf565e2565c370d07fc6aeb95fe3498b4269
#13 4.085   Stored in directory: /tmp/pip-ephem-wheel-cache-_wd1ssof/wheels/c2/3c/b9/1120c2ab4bd82694f7e6f0537dc5b9a085c13e2c69a8d0c76d
#13 4.087 Installing collected packages: boolean-py, binaryornot, setuptools, reuse, python-debian, markupsafe, license-expression, jinja2, chardet
#13 4.110   Attempting uninstall: setuptools
#13 4.110     Found existing installation: setuptools 59.6.0
#13 4.112     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
#13 4.112     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
#13 4.112     Can't uninstall 'setuptools'. No files were found to uninstall.
#13 4.802 Successfully installed binaryornot-0.4.4 boolean-py-4.0 chardet-5.1.0 jinja2-3.1.2 license-expression-30.0.0 markupsafe-2.1.1 python-debian-0.1.49 reuse-1.1.0 setuptools-66.0.0
#13 4.803 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 5.346 Collecting virtualenv
#13 5.384   Downloading virtualenv-20.26.2-py3-none-any.whl (3.9 MB)
#13 5.458      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.9/3.9 MB 54.1 MB/s eta 0:00:00
#13 5.496 Collecting distlib<1,>=0.3.7
#13 5.501   Downloading distlib-0.3.8-py2.py3-none-any.whl (468 kB)
#13 5.507      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 468.9/468.9 KB 112.3 MB/s eta 0:00:00
#13 5.537 Collecting platformdirs<5,>=3.9.1
#13 5.541   Downloading platformdirs-4.2.2-py3-none-any.whl (18 kB)
#13 5.572 Collecting filelock<4,>=3.12.2
#13 5.576   Downloading filelock-3.14.0-py3-none-any.whl (12 kB)
#13 5.664 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 5.828 Successfully installed distlib-0.3.8 filelock-3.14.0 platformdirs-4.2.2 virtualenv-20.26.2
#13 DONE 5.9s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      192448 kB
DirectMap2M:     7147520 kB
DirectMap1G:    11534336 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
downloading https://ci-artifacts.rust-lang.org/rustc-builds-alt/1d43fbbc7348f2bd9260d8532bffa02f5bd2c9ac/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/llvm-1d43fbbc7348f2bd9260d8532bffa02f5bd2c9ac-true/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm
---
   Compiling tidy v0.1.0 (/checkout/src/tools/tidy)
    Finished `release` profile [optimized] target(s) in 27.96s
##[endgroup]
fmt check
##[error]Diff in /checkout/library/core/src/iter/traits/iterator.rs at line 2688:
     #[inline]
     #[unstable(feature = "iterator_try_reduce", reason = "new API", issue = "87053")]
     #[rustc_do_not_const_check]
-    fn try_reduce<R>(&mut self, f: impl FnMut(Self::Item, Self::Item) -> R) -> ChangeOutputType<R, Option<R::Output>>
+    fn try_reduce<R>(
+        &mut self,
+        f: impl FnMut(Self::Item, Self::Item) -> R,
+    ) -> ChangeOutputType<R, Option<R::Output>>
         Self: Sized,
         Self: Sized,
         R: Try<Output = Self::Item, Residual: Residual<Option<Self::Item>>>,
##[error]Diff in /checkout/library/core/src/iter/traits/iterator.rs at line 2953:
     #[inline]
     #[unstable(feature = "try_find", reason = "new API", issue = "63178")]
     #[rustc_do_not_const_check]
-    fn try_find<R>(&mut self, f: impl FnMut(&Self::Item) -> R) -> ChangeOutputType<R, Option<Self::Item>>
+    fn try_find<R>(
+        f: impl FnMut(&Self::Item) -> R,
+        f: impl FnMut(&Self::Item) -> R,
+    ) -> ChangeOutputType<R, Option<Self::Item>>
         Self: Sized,
         Self: Sized,
         R: Try<Output = bool, Residual: Residual<Option<Self::Item>>>,
fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/library/core/src/io/mod.rs" "/checkout/library/core/src/io/borrowed_buf.rs" "/checkout/library/core/src/result.rs" "/checkout/library/core/src/macros/mod.rs" "/checkout/library/core/src/primitive.rs" "/checkout/library/core/src/ops/control_flow.rs" "/checkout/library/core/src/ops/async_function.rs" "/checkout/library/core/src/ops/mod.rs" "/checkout/library/core/src/ops/bit.rs" "/checkout/library/core/src/ops/deref.rs" "/checkout/library/core/src/ops/range.rs" "/checkout/library/core/src/ops/coroutine.rs" "/checkout/library/core/src/ops/function.rs" "/checkout/library/core/src/ops/index_range.rs" "/checkout/library/core/src/ops/arith.rs" "/checkout/library/core/src/ops/drop.rs" "/checkout/library/core/src/ops/try_trait.rs" "/checkout/library/core/src/ops/unsize.rs" "/checkout/library/core/src/ops/index.rs" "/checkout/library/core/src/bool.rs" "/checkout/library/core/src/sync/mod.rs" "/checkout/library/core/src/sync/atomic.rs" "/checkout/library/core/src/sync/exclusive.rs" "/checkout/library/core/src/cmp.rs" "/checkout/library/core/src/panicking.rs" "/checkout/library/core/src/task/ready.rs" "/checkout/library/core/src/task/poll.rs" "/checkout/library/core/src/task/mod.rs" "/checkout/library/core/src/task/wake.rs" "/checkout/library/core/src/cell/once.rs" "/checkout/library/core/src/cell/lazy.rs" "/checkout/library/core/src/async_iter/mod.rs" "/checkout/library/core/src/async_iter/from_iter.rs" "/checkout/library/core/src/async_iter/async_iter.rs" "/checkout/library/core/src/cell.rs" "/checkout/library/core/src/iter/mod.rs" "/checkout/library/core/src/iter/range.rs" "/checkout/library/core/src/iter/traits/mod.rs" "/checkout/library/core/src/iter/traits/marker.rs" "/checkout/library/core/src/iter/traits/unchecked_iterator.rs" "/checkout/library/core/src/iter/traits/iterator.rs" "/checkout/library/core/src/iter/traits/accum.rs" "/checkout/library/core/src/iter/traits/double_ended.rs" "/checkout/library/core/src/iter/traits/exact_size.rs" "/checkout/library/core/src/iter/traits/collect.rs" "/checkout/library/core/src/iter/sources/repeat_with.rs" "/checkout/library/core/src/iter/sources/empty.rs" "/checkout/library/core/src/iter/sources/once.rs" "/checkout/library/core/src/iter/sources/repeat.rs" "/checkout/library/core/src/iter/sources/from_fn.rs" "/checkout/library/core/src/iter/sources/repeat_n.rs" "/checkout/library/core/src/iter/sources/from_coroutine.rs" "/checkout/library/core/src/iter/sources/successors.rs" "/checkout/library/core/src/iter/sources/once_with.rs" "/checkout/library/core/src/iter/adapters/intersperse.rs" "/checkout/library/core/src/iter/adapters/take.rs" "/checkout/library/core/src/iter/adapters/cycle.rs" "/checkout/library/core/src/iter/adapters/mod.rs" "/checkout/library/core/src/iter/adapters/cloned.rs" "/checkout/library/core/src/iter/adapters/filter_map.rs" "/checkout/library/core/src/iter/adapters/take_while.rs" "/checkout/library/core/src/iter/adapters/rev.rs" "/checkout/library/core/src/iter/adapters/chain.rs" "/checkout/library/core/src/error.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Wed Jun 12 18:50:38 UTC 2024
  network time: Wed, 12 Jun 2024 18:50:38 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/library/core/src/iter/traits/iterator.rs
+++ b/library/core/src/iter/traits/iterator.rs
@@ -2080,8 +2080,7 @@ pub trait Iterator {
     fn try_collect<B>(&mut self) -> ChangeOutputType<Self::Item, B>
     where
         Self: Sized,
-        <Self as Iterator>::Item: Try,
-        <<Self as Iterator>::Item as Try>::Residual: Residual<B>,
+        Self::Item: Try<Residual: Residual<B>>,
         B: FromIterator<<Self::Item as Try>::Output>,
     {
         try_process(ByRefSized(self), |i| i.collect())
@@ -2689,12 +2688,13 @@ pub trait Iterator {
     #[inline]
     #[unstable(feature = "iterator_try_reduce", reason = "new API", issue = "87053")]
     #[rustc_do_not_const_check]
-    fn try_reduce<F, R>(&mut self, f: F) -> ChangeOutputType<R, Option<R::Output>>
+    fn try_reduce<R>(
+        &mut self,
+        f: impl FnMut(Self::Item, Self::Item) -> R,
+    ) -> ChangeOutputType<R, Option<R::Output>>
     where
         Self: Sized,
-        F: FnMut(Self::Item, Self::Item) -> R,
-        R: Try<Output = Self::Item>,
-        R::Residual: Residual<Option<Self::Item>>,
+        R: Try<Output = Self::Item, Residual: Residual<Option<Self::Item>>>,
     {
         let first = match self.next() {
             Some(i) => i,
@@ -2956,12 +2956,13 @@ pub trait Iterator {
     #[inline]
     #[unstable(feature = "try_find", reason = "new API", issue = "63178")]
     #[rustc_do_not_const_check]
-    fn try_find<F, R>(&mut self, f: F) -> ChangeOutputType<R, Option<Self::Item>>
+    fn try_find<R>(
+        &mut self,
+        f: impl FnMut(&Self::Item) -> R,
+    ) -> ChangeOutputType<R, Option<Self::Item>>
     where
         Self: Sized,
-        F: FnMut(&Self::Item) -> R,
-        R: Try<Output = bool>,
-        R::Residual: Residual<Option<Self::Item>>,
+        R: Try<Output = bool, Residual: Residual<Option<Self::Item>>>,
     {
         #[inline]
         fn check<I, V, R>(

